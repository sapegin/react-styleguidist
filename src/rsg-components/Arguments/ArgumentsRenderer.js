import React from 'react';
import PropTypes from 'prop-types';
import Argument from 'rsg-components/Argument';
import Styled from 'rsg-components/Styled';

export const styles = ({ space }) => ({
	root: {
		marginBottom: space[2],
		fontSize: 'inherit',
	},
});

export function ArgumentsRenderer({ classes, args }) {
	if (args.length === 0) {
		return null;
	}

	return (
		<div className={classes.root}>
			{args.map(arg => <Argument key={arg.name} {...arg} />)}
		</div>
	);
}

ArgumentsRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	args: PropTypes.arrayOf(
		PropTypes.shape({
			name: PropTypes.string.isRequired,
			type: PropTypes.object,
			description: PropTypes.string,
		})
	).isRequired,
};

export default Styled(styles)(ArgumentsRenderer);
