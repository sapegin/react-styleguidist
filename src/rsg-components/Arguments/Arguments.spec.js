import React from 'react';
import { ArgumentsRenderer } from './ArgumentsRenderer';

it('renderer should render arguments', () => {
	const args = [
		{
			name: 'Foo',
			description: 'Converts foo to bar',
			type: { name: 'Array' },
		},
		{
			name: 'Foo',
		},
	];
	const actual = shallow(<ArgumentsRenderer classes={{}} args={args} />);

	expect(actual).toMatchSnapshot();
});

it('renderer should render nothing for empty array', () => {
	const actual = shallow(<ArgumentsRenderer classes={{}} args={[]} />);

	expect(actual.node).toBe(null);
});
