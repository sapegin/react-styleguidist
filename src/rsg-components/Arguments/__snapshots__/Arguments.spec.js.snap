// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render arguments 1`] = `
<div
  className="root"
>
  <Consumer(Argument)
    description="Converts foo to bar"
    key="Foo"
    name="Foo"
    type={
      Object {
        "name": "Array",
      }
    }
  />
  <Consumer(Argument)
    key="Foo"
    name="Foo"
  />
</div>
`;

exports[`renderer should render heading 1`] = `
<div
  className="root"
>
  <div
    className="headingWrapper"
  >
    <Consumer(Heading)
      level={5}
    >
      Arguments
    </Consumer(Heading)>
  </div>
  <Consumer(Argument)
    key="Foo"
    name="Foo"
  />
</div>
`;
